internal/vds/vds_test.go

```diff
@@ -1064,13 +1062,351 @@ func TestAttribute(t *testing.T) {
 
-		assert.Equalf(
+		assert.InDeltaSlice(
+			t,
+			expected[i],
+			*result,
+			0.000001,
+			"[%s]\nExpected: %v\nActual:   %v",
+			targetAttributes[i],
+			expected[i],
+			*result,
+		)
+	}
+}
+
+/** The vertical domain of the surface is unaligned with the seismic
+ *  but share the samplerate
+ *
+ *  seismic trace   surface window
+ *  -------------   --------------
+ *
+ *            ^
+ *            |
+ *       08ms -
+ *            |       - 9ms
+ *            |       |           - 10ms
+ *            |       |           |           - 11ms
+ *       12ms -       |           |           |
+ *            |       - 13ms      |           |
+ *            |       |           - 14ms      |
+ *            |       |           |           - 15ms
+ *       16ms -       |           |           |
+ *            |       - 17ms      |           |
+ *            |       |           - 18ms      |
+ *            |       |           |           - 19ms
+ *       20ms -       |           |           |
+ *            |       x 21ms      |           |
+ *            |       |           x 22ms      |
+ *            |       |           |           x 23ms
+ *       24ms -       |           |           |
+ *            |       - 25ms      |           |
+ *            |                   - 26ms      |
+ *            |                               - 27ms
+ *       28ms -
+ *            |
+ *            v
+ */
+func TestAttributesUnaligned(t *testing.T) {
+	testCases := []struct{
+		name     string
+		offset   float32
+		expected [][]float32
+	} {
+		{
+			name:   "offset: 0.5",
+			offset: 0.5,
+			expected: [][]float32{ {-2.375}, {0.625}, {-0.875} },
+		},
+		{
+			name:   "offset: 1.0",
+			offset: 1.0,
+			expected: [][]float32{ {-2.250}, {0.750}, {-0.750} },
+		},
+		{
+			name:   "offset: 2.0",
+			offset: 2.0,
+			expected: [][]float32{ {-2.000}, {1.000}, {-0.500} },
+		},
+		{
+			name:   "offset: 3.0",
+			offset: 3.0,
+			expected: [][]float32{ {-1.750}, {1.250}, {-0.250} },
+		},
+		{
+			name:   "offset: 3.5",
+			offset: 3.5,
+			expected: [][]float32{ {-1.625}, {1.375}, {-0.125} },
+		},
+	}
+
+	const fill       = float32(-999.25)
+	const above      = float32(8)
+	const below      = float32(4)
+	const samplerate = float32(4)
+
+	targetAttributes := []string{"min", "max", "mean"}
+	interpolationMethod, _ := GetInterpolationMethod("nearest")
+	
+	for _, testCase := range testCases {
+		horizon := [][]float32{{ 20 + testCase.offset} }
+
+		handle, _ := NewVDSHandle(samples10)
+		defer handle.Close()
+		buf, err := handle.GetAttributes(
+			horizon,
+			samples10_grid.xori,
+			samples10_grid.yori,
+			samples10_grid.xinc,
+			samples10_grid.yinc,
+			samples10_grid.rotation,
+			fill,
+			above,
+			below,
+			samplerate,
+			targetAttributes,
+			interpolationMethod,
+		)
+		if err != nil {
+			t.Errorf("Failed to fetch horizon, err: %v", err)
+		}
+
+		if len(buf) != 3 {
+			t.Errorf("Incorrect number of attributes returned")
+		}
```

Comment:
Btw think that here and in all tests below we could benefit from using testify. Instead of 3 lines per check we could have 1, which hopefully would reduce noise.